File: 5월 2주차/Jhanoo/03/[PGS - Lv2] 연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        // 1,000 * 1,000 = 1,000,000
        HashSet<Integer> s = new HashSet<>();
        
        int n = elements.length;
        for (int i = 0; i < n; i++) {
            for (int len = 1; len <= n; len++) {
                int sum = 0;
                for (int j = 0; j < len; j++) 
                    sum += elements[(i + j) % n];
                
                s.add(sum);
            }
        }
        
        return s.size();
    }
}

File: 2026 07월/7월 3주차/JeongEon8/02/[PGS - Lv2] 연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    int start = 0, end = 0;
    set<int> sums;
    
    for(int i = 0; i < elements.size(); ++i)
    {
        int sum = 0;
        for(int j = i; j < i+elements.size(); ++j)
        {
            int idx = j % elements.size();
            sum += elements[idx];
            sums.emplace(sum);
        }
    }
    
    answer = sums.size();
    
    return answer;
}
